### ISSUE TITLE
New transaction_commitment bug & general observation 

### ISSUE THREAD

**Issue by rorads** [devinit/D-Portal#509]
Preamble: I can see that several great UX enhancements have gone into D-Portal recently - good work, thank you. Now to nit-pick about something :)

I've just checked out this activity published by DFID: http://d-portal.org/ctrack.html#view=act&aid=GB-1-204252-132

Looking at devtracker's back end, I can see that there is one commitment in this activity, for £1.3m. 

If I look at the activity page linked above and scroll to transactions, where there used to be a list of commitments, I now see this graph:

![Imgur](https://i.imgur.com/0H6riIa.png)

There's clearly an issue with the x-axis, which shows the same date for each increment, so it would be good to get this fixed.

More generally it seems that this graph assumes multiple data points, and tries to infer a line graph. I'd suggest that this isn't the best option, as there will be many activities with only one or two commitments and the line will mean very little (as it does in the example above). 

I also think that this kind of graph implies that the area under the line is meaningful, which I don't think it is with transactions. I'd suggest a scatter graph, possibly with a trend line, but even that implies that the trend is significant (or even normative), which again could be misleading.

**Comment by k8hughes** [comment]
I've added the "bug" label for the x-axis the other part of what is raise here involves more of the discussion.

This page shows how the graph works well went there are multiple transactions, however when there is only one transaction on a graph it does look odd.  It would be good to look at some options for this case.

http://d-portal.org/ctrack.html?publisher=46004#view=act&aid=46004-46268-002-LN3

**Comment by notshi** [comment]
Thanks for raising this.
We went through some iterations before we were happy with this current version.

The DFID activity now also lists a disbursement so the graph looks like this:
![Screenshot-20190318190609-930x315](https://user-images.githubusercontent.com/1515961/54556516-ecf40e80-49b0-11e9-83ee-b0272c2f6260.png)

The graph does not assume multiple data points, we have single transactions stretch across the lifetime of the transaction data. This gives context to the main graph that displays all transaction types together.

The problem here is displaying a graph when there is little data. Previously, we've removed graphs when there is only one data point but it feels wrong as we prefer not to have exceptions for things.

Ultimately, the question is not about making an activity with little data look good but make data good.
